File: stack.cpp
#include <stdio.h>
#include <stdlib.h>

typedef struct node {
    int data;
     node *prev; 
} node;

node *first ;
node *last ;

void push(int x) {
    node *temp = (node *)malloc(sizeof(node));
    temp->data = x;
    temp->prev = first; // Düðümün bir sonraki düðümü, mevcut "first" düðümünü gösterir.

    first = temp;
}

int pop() {
    int postala;
    postala = first->data;
    node *temp = first;
    first = first->prev;
    free(temp);

    return postala;
}


void listele() {
    node *temp = first;
    while (temp) {
        printf(" %d ", temp->data);
        temp = temp->prev;
    }
    printf("\n");
}

int main() {
    push(1);
    push(2);
    push(3);
    pop();
    listele();

    return 0;
}

File: stack.py
class Node:
    def __init__(self, data=None, next=None):
        self.data = data
        self.next = next

class Stack:
    def __init__(self):
        self.head = None

    def add(self, data):
        node = Node(data, self.head)
        self.head = node

    def pop(self):
        if self.head:
            self.head = self.head.next
        else:
            print("stack is empty")

    def print(self):
        itr = self.head
        while itr:
            print("|")
            print(itr.data)
            itr = itr.next
        print("---------------------")

ll = Stack()  

ll.add(4)
ll.print()
ll.add(2)
ll.print()
ll.add(3)
ll.print()
ll.pop()
ll.print()
ll.pop()
ll.print()
ll.pop()
ll.print()
ll.pop()
ll.print()

        
